.github/workflows/node.js.yml

```diff
@@ -0,0 +1,33 @@
+name: ci
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - uses: actions/checkout@v2
+    - name: setup
+      uses: actions/setup-node@v1
```

Comment:
Should we specify a node version here? Looks like by default at this time its `v10.22.0` which is fine because thats our minimum required. We should just keep an eye on it though (https://github.com/actions/setup-node) in case they bump the default version upstream and we want to keep our tests running on v10

Reply:
I think we want to keep it running on the latest 10.x, not necessarily older minor versions. We can also do a matrix of 10.x, 12.x and 14.x if we want.

Reply:
Demo for the matrix: https://github.com/tuxcanfly/hsd/pull/21

I think latest versions make sense because there might be security or other updates which we might want to support.

Reply:
While on the topic of matrix, we should checkout running across OS'es like macos and windows. Seems like GitHub supports these: 

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions

Also, maybe we should be running `test-browser` for browser tests across each matrix item.

It adds to the build-time but would prevent recurring cross-platform bugs we encounter (windows).

Reply:
The nodejs matrix is super cool - github says it only took 3m 55s to complete, that's more than acceptable I think. I think OS and browser testing would be great too - if it's too much work, could we just run those extra tests on version-tagging commits or something? (for releases, not just every PR)

Reply:
Sounds good. Even a manual trigger for those wouldn't be too bad.

That could go into another PR since it needs updating the workflow for cross-platform builds.

For now, I suggest we should adopt the node 10.x, 12.x and 14.x build matrix.

Reply:
yup agreed, lets add that then try to get ACKs

Reply:
OK, done.